### PR TITLE
Rrojo/plugin mgr cfg

### PR DESCRIPTION
**Pull request by r-rojo**
Merge master into feature/plugin-mgr-config

### CODE REVIEW

**Comment by r-rojo** [comment]
I will create a PR to merge this feature branch into master (without merging from master first).

File: CMakeLists.txt [hunk 1]
```diff
@@ -221,7 +222,7 @@ set(CPACK_PACKAGE_VERSION_MAJOR "${INTEL_FPGA_API_VER_MAJOR}")
 set(CPACK_PACKAGE_VERSION_MINOR "${INTEL_FPGA_API_VER_MINOR}")
 set(CPACK_PACKAGE_VERSION_PATCH "${INTEL_FPGA_API_VER_REV}")
 set(CPACK_PACKAGE_VERSION ${INTEL_FPGA_API_VERSION})
-set(CPACK_PACKAGE_RELEASE 1)
+set(CPACK_PACKAGE_RELEASE 2)
```
Comment (lzhan55) [inline CMakeLists.txt]: What is this release number about? Not OPAE?

File: libbitstream/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,69 @@
+## Copyright(c) 2019, Intel Corporation
```
Comment (lzhan55) [inline libbitstream/CMakeLists.txt]: Are we adding this bitstream as a separate shared library which was previously part of xfpga library?

File: libopae/pluginmgr.c [hunk 3]
```diff
@@ -62,6 +62,9 @@ static platform_data platform_data_table[] = {
 	{ 0x8086, 0x09c4, "libxfpga.so", 0 },
 	{ 0x8086, 0x09c5, "libxfpga.so", 0 },
 	{ 0x8086, 0x0b2b, "libxfpga.so", 0 },
+	{ 0x8086, 0x0b2c, "libxfpga.so", 0 },
```
Comment (lzhan55) [inline libopae/pluginmgr.c]: These new vendor ids are for darby/vista creek etc.?

File: libbitstream/CMakeLists.txt (hunk 2 above)
Comment (r-rojo) [reply libbitstream/CMakeLists.txt]: This is a merge from master.

File: libopae/pluginmgr.c (hunk 3 above)
Comment (r-rojo) [reply libopae/pluginmgr.c]: This is a merge from master.

**Comment by r-rojo** [comment]
I did not hand merge any files. This is just a straight merge.

File: CMakeLists.txt [hunk 4]
```diff
@@ -233,10 +234,56 @@ endif()
 
 if("${CPACK_GENERATOR}" STREQUAL "DEB")
 include(packaging)
+
+#list of components to be included in the package
+set(CPACK_COMPONENTS_ALL 
+	toolfpgaconf
```
Comment (lzhan55) [inline CMakeLists.txt]: Is this CPACK_COMPONENTS_ALL used somewhere?

Comment (nakulkorde) [reply CMakeLists.txt]: That variable is to specify what components need to be packaged in the debian packages.

File: libopae/plugins/xfpga/metrics/bmc/CMakeLists.txt [hunk 5]
```diff
@@ -25,31 +25,43 @@
...
+    ${CMAKE_THREAD_LIBS_INIT})
+
+add_library(staticbmc STATIC ${SRC})
 
+set_property(TARGET staticbmc PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+add_library(modbmc MODULE ${SRC})
+
+target_link_libraries(modbmc
+    safestr
```
Comment (lzhan55) [inline libopae/plugins/xfpga/metrics/bmc/CMakeLists.txt]: What if the difference between libbmc.so and libmodbmc.so?

**Pull request merged**